src/resources/views/crud/filters/date_range.blade.php

```diff
@@ -58,84 +78,93 @@ class="nav-item dropdown {{ Request::get($filter->name)?'active':'' }}">
   			} else {
-  				//this change to empty string,because addOrUpdateUriParameter method just judgment string
   				var value = '';
   			}
-			var parameter = '{{ $filter->name }}';
 
+            var parameter = '{{ $filter->name }}';
 	    	// behaviour for ajax table
 			var ajax_table = $('#crudTable').DataTable();
 			var current_url = ajax_table.ajax.url();
 			var new_url = addOrUpdateUriParameter(current_url, parameter, value);
-
 			// replace the datatables ajax url with new_url and reload it
 			new_url = normalizeAmpersand(new_url.toString());
 			ajax_table.ajax.url(new_url).load();
-
 			// add filter to URL
 			crud.updateUrl(new_url);
-
 			// mark this filter as active in the navbar-filters
 			if (URI(new_url).hasQuery('{{ $filter->name }}', true)) {
 				$('li[filter-key={{ $filter->key }}]').removeClass('active').addClass('active');
-			}
-			else
-			{
+			} else {
 				$('li[filter-key={{ $filter->key }}]').trigger('filter:clear');
 			}
   		}
 
 		jQuery(document).ready(function($) {
-			var dateRangeInput = $('#daterangepicker-{{ $filter->key }}').daterangepicker({
-				timePicker: false,
-		        ranges: {
-		            'Today': [moment().startOf('day'), moment().endOf('day')],
-		            'Yesterday': [moment().subtract(1, 'days'), moment().subtract(1, 'days')],
-		            'Last 7 Days': [moment().subtract(6, 'days'), moment()],
-		            'Last 30 Days': [moment().subtract(29, 'days'), moment()],
-		            'This Month': [moment().startOf('month'), moment().endOf('month')],
-		            'Last Month': [moment().subtract(1, 'month').startOf('month'), moment().subtract(1, 'month').endOf('month')]
-		        },
-				@if ($filter->currentValue)
-		        startDate: moment("{{ $start_date }}"),
-		        endDate: moment("{{ $end_date }}"),
-				@endif
-				alwaysShowCalendars: true,
-				autoUpdateInput: true
-			});
+			var dateRangeInput = $('#daterangepicker-{{ $filter->key }}');
+
+            $config = dateRangeInput.data('bs-daterangepicker');
+
+            $ranges = $config.ranges;
+            $config.ranges = {};
+
+            //if developer configured ranges we convert it to moment() dates.
+            for (var key in $ranges) {
+                if ($ranges.hasOwnProperty(key)) {
+                    $config.ranges[key] = $.map($ranges[key], function($val) {
+                        return moment($val);
+                    });
+                }
+            }
+
+            if($config.startDate) {
+                $config.startDate = moment($config.startDate, $config.format);
+            }else{
+                $config.startDate = moment(moment().format($config.format));
+            }
 
-			dateRangeInput.on('apply.daterangepicker', function(ev, picker) {
+            if($config.endDate) {
+                $config.endDate = moment($config.endDate, $config.format);
+            }else{
+                $config.endDate = moment(moment().format($config.format));
+            }
```

Comment:
@pxpm what do you think about adding `startDate` and `endDate` has current date on the configs?

```...
		'startDate' => \Carbon\Carbon::now()->toDateTimeString(),
		'endDate' => \Carbon\Carbon::now()->toDateTimeString(),
```

This way, we could avoid these two else statements, avoiding the mess of creating dates with the user date format.

Reply:
Also, when doing `moment($config.endDate, $config.format);` you are relying on the format user choose, which probably is different from the default format Carbon outputs.

Basically I thing we don't need to pass the date format when creating moment instance. Because the date comes from Carbon, it's ISO formatted.

Reply:
@promatik thanks for the heads up. Will refactor accordingly.

Reply:
@pxpm btw, to have the weekdays and months translated, you need to do `moment.locale('pt');` before everything.
You will also need to use `packages/moment/min/moment-with-locales.min.js` instead of the default `moment.min.js`.